src/main/java/com/powsybl/network/hypothesis/HypothesisUtils.java

```diff
@@ -0,0 +1,133 @@
+/**
+ * Copyright (c) 2021, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.network.hypothesis;
+
+import com.powsybl.iidm.network.*;
+
+import java.util.function.BiConsumer;
+
+/**
+ * Some useful utility methods to create network hypotheses.
+ *
+ * @author Miora Ralambotiana <miora.ralambotiana at rte-france.com>
+ */
+public final class HypothesisUtils {
+
+    /**
+     * Split a given line and create a fictitious voltage level at the junction.<br>
+     * The line is considered split in half such as this:<br>
+     * <code>r1 = 0.5 * r</code><br>
+     * <code>r2 = 0.5 * r</code><br>
+     * If the line is between two BUS-BREAKER voltage levels, the fictitious voltage level will be BUS-BREAKER and the two new lines
+     * will be linked to a fictitious bus in this voltage level.<br>
+     * If the line is between the two NODE-BREAKER voltage levels, the fictitious voltage level will be NODE_BREAKER and the
+     * two lines will be linked via an internal connection between two nodes in this voltage level.
+     */
+    public static void createVoltageLevelOnLine(Line line) {
+        createVoltageLevelOnLine(0.5, 0.5, 0.5, 0.5, 0.5, 0.5, line);
+    }
+
+    /**
+     * Split a given line and create a fictitious voltage level at the junction.<br>
+     * The characteristics of the two new lines respect the given ratios such as this:<br>
+     * <code>r1 = rdp * r</code><br>
+     * <code>r2 = (1 - rdp) * r</code><br>
+     * If the line is between two BUS-BREAKER voltage levels, the fictitious voltage level will be BUS-BREAKER and the two new lines
+     * will be linked to a fictitious bus in this voltage level.<br>
+     * If the line is between the two NODE-BREAKER voltage levels, the fictitious voltage level will be NODE_BREAKER and the
+     * two lines will be linked via an internal connection between two nodes in this voltage level.
+     */
+    public static void createVoltageLevelOnLine(double rdp, double xdp, double g1dp, double b1dp, double g2dp, double b2dp,
+                                                Line line) {
+        TopologyKind topologyKind = line.getTerminal1().getVoltageLevel().getTopologyKind();
+        if (topologyKind != line.getTerminal2().getVoltageLevel().getTopologyKind()) {
+            throw new AssertionError();
+        }
+        Network network = line.getNetwork();
+        Substation substation = network.newSubstation()
```

Comment:
More a side question than a comment: we create here a fictitious substation, but once the v4.4.0 of core is released we will create the voltage level directly from the network?

Reply:
Yes, the voltage level will be created directly from the network